docs/source/package_reference/tuners.md

```diff
@@ -1,43 +1,27 @@
-<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
-rendered properly in your Markdown viewer.
--->
-
-# Tuners
-
-Each tuner (or PEFT method) has a configuration and model.
-
-## LoRA
-
-For finetuning a model with LoRA.
-
-[[autodoc]] LoraConfig
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
 
-[[autodoc]] LoraModel
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
 
-[[autodoc]] tuners.lora.LoraLayer
+http://www.apache.org/licenses/LICENSE-2.0
 
-[[autodoc]] tuners.lora.Linear
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
 
-## P-tuning
-
-[[autodoc]] tuners.p_tuning.PromptEncoderConfig
-
-[[autodoc]] tuners.p_tuning.PromptEncoder
-
-## Prefix tuning
-
-[[autodoc]] tuners.prefix_tuning.PrefixTuningConfig
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
 
-[[autodoc]] tuners.prefix_tuning.PrefixEncoder
+-->
 
-## Prompt tuning
+# Tuners
 
-[[autodoc]] tuners.prompt_tuning.PromptTuningConfig
+A tuner (or adapter) is a module that can be plugged into a `torch.nn.Module`. [`BaseTuner`] base class for other tuners and provides shared methods and attributes for preparing an adapter configuration and replacing a target module with the adapter module. [`BaseTunerLayer`] is a tuner mixin offering methods and attributes for managing adapters such as merging and unmerging, and activating and disabling adapters.
```

Comment:
```suggestion
A tuner (or adapter) is a module that can be plugged into a `torch.nn.Module`. [`BaseTuner`] base class for other tuners and provides shared methods and attributes for preparing an adapter configuration and replacing a target module with the adapter module. [`BaseTunerLayer`] is a base class for adapter layers. It offers methods and attributes for managing adapters such as activating and disabling adapters.
```

`BaseTunerLayer` is not really a mixin, since it has abstract methods. Also, it doesn't implement (un)merging, these need to be implemented by child classes if the corresponding method supports it.